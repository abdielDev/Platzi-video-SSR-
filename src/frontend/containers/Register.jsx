import React, { useState } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { registerUser } from '../actions';
import '../assets/styles/components/Register.scss';

const Register = (props) => {

  const [form, setValues] = useState({
    email: '',
    name: '',
    password: '',
  });

  const handleInput = (event) => {
    setValues({
      ...form,
      [event.target.name]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    props.registerUser(form, '/login');
  };

  return (
    <section className="register">
      <section className="register__container">
        <h2>Regístrate</h2>
        <form className="register__container--form" action="" onSubmit={handleSubmit}>
          <input
            name="name"
            aria-label="Nombre"
            className="login__input"
            type="text"
            placeholder="Nombre"
            onChange={handleInput}
            required
          />
          <input
            name="email"
            aria-label="Correo"
            className="login__input"
            type="text"
            placeholder="Correo"
            onChange={handleInput}
            required
          />
          <input
            name="password"
            aria-label="Contraseña"
            className="login__input"
            type="password"
            placeholder="Contraseña"
            onChange={handleInput}
            required
          />
          <button type="submit" className="button">Regístrarme</button>
        </form>
        <p className="register__container--register"><Link to="/login">Iniciar sesión</Link></p>
      </section>
    </section>
  );
};

const mapDispatchToProps = {
  registerUser,
};

export default connect(null, mapDispatchToProps)(Register);
