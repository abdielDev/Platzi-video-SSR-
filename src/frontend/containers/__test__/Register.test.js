import React from 'react';
import { create } from 'react-test-renderer';
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import ProviderMock from "../../__mocks__/providerMock";
import Register from "../Register";

configure({ adapter: new Adapter() });
try {
    describe('Register component', () => {
        test('Match Snapshot', () => {
            const register = create(
                <ProviderMock>
                    <Register/>
                </ProviderMock>,
            );
            expect(register.toJSON()).toMatchSnapshot();
        });

        it('Calls and executes preventDefautl function onSubmit form', () => {
            const preventDefault = jest.fn();
            const wrapper = mount(
                <ProviderMock>
                    <Register/>
                </ProviderMock>,
            );
            wrapper.find('form').simulate('submit', {preventDefault});
            expect(preventDefault).toHaveBeenCalledTimes(1);
            wrapper.unmount();
        });
    });
} catch (e) {
    console.error(e)
}
