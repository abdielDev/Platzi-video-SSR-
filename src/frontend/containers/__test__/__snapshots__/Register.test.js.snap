// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component Match Snapshot 1`] = `
Array [
  <header
    className="header isRegister"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Platzi vídeo logo"
        className="header__img"
        src="test-file-stubs"
      />
    </a>
    <div
      className="header__menu"
    >
      <div
        className="header__menu--profile"
      >
        <img
          alt="Avatar logo"
          src="test-file-stubs"
        />
        <p>
          Perfil
        </p>
      </div>
      <ul>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Iniciar sesión
          </a>
        </li>
      </ul>
    </div>
  </header>,
  <section
    className="register"
  >
    <section
      className="register__container"
    >
      <h2>
        Regístrate
      </h2>
      <form
        action=""
        className="register__container--form"
        onSubmit={[Function]}
      >
        <input
          aria-label="Nombre"
          className="login__input"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          required={true}
          type="text"
        />
        <input
          aria-label="Correo"
          className="login__input"
          name="email"
          onChange={[Function]}
          placeholder="Correo"
          required={true}
          type="text"
        />
        <input
          aria-label="Contraseña"
          className="login__input"
          name="password"
          onChange={[Function]}
          placeholder="Contraseña"
          required={true}
          type="password"
        />
        <button
          className="button"
          type="submit"
        >
          Regístrarme
        </button>
      </form>
      <p
        className="register__container--register"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Iniciar sesión
        </a>
      </p>
    </section>
  </section>,
]
`;
